lib/datadog/statsd.rb

```diff
@@ -52,6 +52,7 @@ def write(message)
         bad_socket = !@socket_path.nil? && (
           boom.is_a?(Errno::ECONNREFUSED) ||
           boom.is_a?(Errno::ECONNRESET) ||
+          boom.is_a?(Errno::ENOTCONN) ||
```

Comment:
It might actually be better to not add this and instead relax the retry conditions on this line https://github.com/DataDog/dogstatsd-ruby/blob/570f3c097f55a44e7d3db781fcc896435ac769cf/lib/datadog/statsd.rb#L64
to just 



```ruby
        if retries <= 1
```

In our case as well as I'm sure in many others, we'd _much_ rather immediately retry establishing the socket and resending the message rather than losing the message and having to wait for the next one for the socket to re-establish itself.

Reply:
Good idea, I changed the logic to retry the same message upon ENOTCONN